### PR TITLE
Feature/GitHub actions

### PR DESCRIPTION
**Pull request by MikeHV14**
## Description

adding the driven adapter for the github devops platform

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] The pull request is complete according to the guide of [contributing](https://github.com/bancolombia/devsecops-engine-tools/blob/trunk/docs/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: tools/devsecops_engine_tools/engine_utilities/github/models/GithubPredefinedVariables.py
Comment (Gaviria9601) [inline tools/devsecops_engine_tools/engine_utilities/github/models/GithubPredefinedVariables.py:1]: I don't see the use of github variables. Please check the link https://docs.github.com/en/actions/migrating-to-github-actions/automated-migrations/migrating-from-azure-devops-with-github-actions-importer#environment-variable-mapping

File: tools/devsecops_engine_tools/engine_core/src/infrastructure/driven_adapters/github/github_actions.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        utils_github = GithubApi(
+            personal_access_token=SystemVariables.GH_AccessToken.value()
+        )
+
+        git_client = utils_github.get_github_connection()
+        json_config = utils_github.get_remote_json_config(git_client, owner, repository, path)
+
+        return json_config
+
+    def message(self, type, message):
```
Comment (Gaviria9601) [inline tools/devsecops_engine_tools/engine_core/src/infrastructure/driven_adapters/github/github_actions.py:40]: Please review the worflow commands: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions

File: tools/devsecops_engine_tools/engine_core/src/infrastructure/driven_adapters/github/github_actions.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        formats = {
+            "succeeded": f"{self.OKGREEN}{message}{self.ENDC}",
+            "info": f"{self.BOLD}{message}{self.ENDC}",
+            "warning": f"{self.WARNING}{message}{self.ENDC}",
+            "error": f"{self.FAIL}{message}{self.ENDC}"
+        }
+        return formats.get(type, message)
+
+    def result_pipeline(self, type):
```
Comment (Gaviria9601) [inline tools/devsecops_engine_tools/engine_core/src/infrastructure/driven_adapters/github/github_actions.py:49]: Please review the worflow commands: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions

File: tools/devsecops_engine_tools/engine_utilities/github/models/GithubPredefinedVariables.py
Comment (Gaviria9601) [inline tools/devsecops_engine_tools/engine_utilities/github/models/GithubPredefinedVariables.py:1]: Delete .gitkeep

**Review by Gaviria9601 — COMMENTED** [review]
Please, review the comments

**Review by Gaviria9601 — COMMENTED** [review]
Up versión in tools/devsecops_engine_tools/version.py and update the documentation in readme to use --platform_devops github

File: tools/devsecops_engine_tools/engine_core/src/infrastructure/driven_adapters/github/github_actions.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        results = {
+            "failed": f"{self.FAIL}{self.ICON_FAIL}Failed{self.ENDC}",
+            "succeeded": f"{self.OKGREEN}{self.ICON_SUCCESS}Succeeded{self.ENDC}"
+        }
+        return results.get(type)
+
+    def get_source_code_management_uri(self):
```
Comment (Gaviria9601) [inline tools/devsecops_engine_tools/engine_core/src/infrastructure/driven_adapters/github/github_actions.py:56]: I don't see the use of github variables. Please check the link https://docs.github.com/en/actions/migrating-to-github-actions/automated-migrations/migrating-from-azure-devops-with-github-actions-importer#environment-variable-mapping and use GithubPredefinedVariables

File: tools/devsecops_engine_tools/engine_core/src/infrastructure/driven_adapters/github/github_actions.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        return os.environ.get("GH_SOURCE_CODE_MANAGEMENT_URI")
+
+    def get_base_compact_remote_config_url(self, remote_config_repo):
```
Comment (Gaviria9601) [inline tools/devsecops_engine_tools/engine_core/src/infrastructure/driven_adapters/github/github_actions.py:59]: I don't see the use of github variables. Please check the link https://docs.github.com/en/actions/migrating-to-github-actions/automated-migrations/migrating-from-azure-devops-with-github-actions-importer#environment-variable-mapping and use GithubPredefinedVariables

**Review by luarredo — APPROVED** [review]

**Pull request merged**